exchanges/orderbook/orderbook_test.go

```diff
@@ -263,4 +266,62 @@ func TestProcessOrderbook(t *testing.T) {
 	if a != 200 && b != 84000 {
 		t.Fatal("Test failed. TestProcessOrderbook CalculateTotalsBids incorrect values")
 	}
+
+	type quick struct {
+		Name string
+		P    pair.CurrencyPair
+		Bids []Item
+		Asks []Item
+	}
+
+	var testArray []quick
+
+	_ = rand.NewSource(time.Now().Unix())
+
+	var wg sync.WaitGroup
+
+	for i := 0; i < 500; i++ {
+		wg.Add(1)
+		go func() {
+			newName := "Exchange" + strconv.FormatInt(rand.Int63(), 10)
+			newPairs := pair.NewCurrencyPair("BTC"+strconv.FormatInt(rand.Int63(), 10),
+				"USD"+strconv.FormatInt(rand.Int63(), 10))
+
+			asks := []Item{{Price: rand.Float64(), Amount: rand.Float64()}}
+			bids := []Item{{Price: rand.Float64(), Amount: rand.Float64()}}
+			base := Base{
+				Pair:         newPairs,
+				CurrencyPair: newPairs.Pair().String(),
+				Asks:         asks,
+				Bids:         bids,
+			}
+
+			ProcessOrderbook(newName, newPairs, base, Spot)
+			testArray = append(testArray, quick{Name: newName, P: newPairs, Bids: bids, Asks: asks})
```

Comment:
Races should only occur on the track :P.  Could you please add a mutex around this append?